bee-api/bee-rest-api/src/endpoints/routes/api/v2/submit_message.rs

```diff
@@ -78,40 +81,39 @@ pub(crate) async fn submit_message<B: StorageBackend>(
     rest_api_config: RestApiConfig,
     protocol_config: ProtocolConfig,
 ) -> Result<impl Reply, Rejection> {
-    let network_id_v = &value["networkId"];
-    let parents_v = &value["parentMessageIds"];
-    let payload_v = &value["payload"];
-    let nonce_v = &value["nonce"];
+    let protocol_version_json = &value["protocolVersion"];
+    let parents_json = &value["parentMessageIds"];
+    let payload_json = &value["payload"];
+    let nonce_json = &value["nonce"];
 
-    // parse the values, take care of missing fields, build the message and submit it to the node for further
-    // processing
+    // Tries to build a `Message` from the given JSON.
+    // If some fields are missing, it tries to auto-complete them.
 
-    let network_id = if network_id_v.is_null() {
-        network_id.1
-    } else {
-        network_id_v
-            .as_str()
-            .ok_or_else(|| {
-                reject::custom(CustomRejection::BadRequest(
-                    "invalid network id: expected an u64-string".to_string(),
-                ))
-            })?
-            .parse::<u64>()
-            .map_err(|_| {
-                reject::custom(CustomRejection::BadRequest(
-                    "invalid network id: expected an u64-string".to_string(),
-                ))
-            })?
-    };
+    if !protocol_version_json.is_null() {
+        let parsed = protocol_version_json.as_u64().ok_or_else(|| {
+            reject::custom(CustomRejection::BadRequest(
+                "invalid protocol version: expected an unsigned integer < 256".to_string(),
+            ))
+        })? as u8;
 
-    let parents: Vec<MessageId> = if parents_v.is_null() {
+        // TODO: is this check really necessary here considering that the "node processing pipe" anyways checks the
+        // field for incoming messages?
```

Comment:
Maybe make this a NOTE rather than a TODO, because I feel like this could become something that keeps hanging around for all eternity. You could argue that this additional check is probably not necessary due to the processing pipe, but to make the crate more self-contained/self-sufficient (not relying on other crates to perform the check) it is performed anyway.

Reply:
When the user submits raw bytes we don't even do the checks in the API.

Reply:
I think the protocol version and network ID are different regarding this. If the protocol version is wrong, the JSON deserialization will most probably fail so we should even try. Network ID doesn't influence deserialization.

Reply:
>  If the protocol version is wrong, the JSON deserialization will most probably fail
Why is the JSON deserialization failing? Not sure I follow